connection_test.go

```diff
@@ -519,6 +519,91 @@ func TestConnectionFetch(t *testing.T) {
 	}
 }
 
+func TestTrimRecordBatches(t *testing.T) {
+	resp := &proto.FetchResp{
+		CorrelationID: 2,
+		Topics: []proto.FetchRespTopic{
+			{
+				Name: "foo",
+				Partitions: []proto.FetchRespPartition{{
+					ID:             1,
+					Err:            nil,
+					TipOffset:      20,
+					MessageVersion: 2,
+					RecordBatches: []*proto.RecordBatch{{
+						FirstOffset: 4,
+						Records: []*proto.Record{{
+							OffsetDelta: 0,
+							Key:         []byte("f"),
+							Value:       []byte("first"),
+						}, {
+							OffsetDelta: 1,
```

Comment:
@e-max please also review this test data carefully.  I'm only really making guesses at how the `OffsetDelta`s work.  I haven't seen a multi-record batch from our test server.